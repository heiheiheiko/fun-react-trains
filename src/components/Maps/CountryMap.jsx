import React from "react";
import PropTypes from 'prop-types';
import { Geographies, Geography } from "react-simple-maps";
import Map from './Map';
import Country from './Country';

function CountryMap(props) {
  return (
    <Map>
      <Geographies geography="/maps/world-50m.json">
        {(geographies, projection) =>
          geographies.map((geography, i) =>
            props.countries.includes(geography.id) && (
              <Country geography={geography} projection={projection} key={i} />
            )
          )
        }
      </Geographies>
    </Map>
  );
}

CountryMap.propTypes = {
  countries: PropTypes.array,
};

export default CountryMap;