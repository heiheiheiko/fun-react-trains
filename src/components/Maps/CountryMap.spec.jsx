import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer'
import SpecContainer from '../../../__specs__/helpers/SpecIntelContainer';

import CountryMap from './CountryMap';

jest.mock('./Country', () => ()=> "Country");
jest.mock('./Map', () => ()=> "Map");

describe('<CountryMap>', () => {
  it('could be renderd', () => {
    const mountedSpecContainer = mount(<SpecContainer><CountryMap/></SpecContainer>);
    expect(mountedSpecContainer.find(CountryMap).length).toEqual(1);
  });

  it('matched the snapshot', () => {
    const specContainer = <SpecContainer><CountryMap/></SpecContainer>;
    const renderedValue =  renderer.create(specContainer).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });
});