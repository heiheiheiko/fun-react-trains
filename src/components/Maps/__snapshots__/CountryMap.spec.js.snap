// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryMap> matched the snapshot 1`] = `
<div>
   
  Map
   
</div>
`;
